### ISSUE TITLE
Delete Image Error

### ISSUE THREAD

**Issue by tawsif93** [aberezkin/ng2-image-upload#208]
When Image Limit completed, then only the clear button is visible. If I remove all the images or A single image, I'm getting an error from the component. Also, delete event is not fired.

`UploadDocumentComponent.html:18 ERROR TypeError: Cannot read property 'nativeElement' of undefined
    at ImageUploadComponent.deleteAll (image-upload.component.js:85)
    at Object.eval [as handleEvent] (ImageUploadComponent.html:27)
    at handleEvent (core.js:13589)
    at callWithDebugContext (core.js:15098)
    at Object.debugHandleEvent [as handleEvent] (core.js:14685)
    at dispatchEvent (core.js:10004)
    at eval (core.js:10629)
    at HTMLButtonElement.eval (platform-browser.js:2628)
    at ZoneDelegate.invokeTask (zone.js:421)
    at Object.onInvokeTask (core.js:4751)`

**Comment by majidnk01** [comment]
Yes same issue here... I don't know it's good practice or not (of course not :) ) but it solved my issue by commenting out this line in  **ImageUploadComponent.prototype.deleteFile** 

`        this.inputElement.nativeElement.value = '';
`
in 

` /node_modules/angular2-image-upload/lib/image-upload/image-upload.component.js `

**Comment by tawsif93** [comment]
@majidnk01  Obviously its not a good approach, when you are working remote. It will solve my current development problem but will cause problem for others.

**Comment by AntiLoxy** [comment]
Same issue on Angular 6 with your last version.

**Comment by ramajd** [comment]
I found a workaround for this issue, just call contents of `deleteAll()` method directly in your code:

```ts
...
@ViewChild('logoUploader') logoUploader: ImageUploadComponent;
...
// this.logoUploader.deleteAll();
this.logoUploader.files.forEach(f => this.logoUploader.removed.emit(f));
this.logoUploader.files = [];
this.logoUploader.fileCounter = 0;
...

```

**Comment by MarwaAbuEssa** [comment]
the same here

**Comment by LFeh** [comment]
Publish it on npm, please

**Comment by tawsif93** [comment]
@aberezkin , @UncleDave  Can you please **publish** this update?

**Comment by xpvo** [comment]
I'm having the same issue and it hasn't been published on npm yet!

**Comment by Taylorsuk** [comment]
Same here +1

**Comment by josemmarneca** [comment]
The problem is the parameter [max]="xxxxx"
I solve my problem, remove the parameter and add the parameter [beforeUpload]="onBeforeUpload".
private fileCounter = 0;

onBeforeUpload = (metadata: UploadMetadata) => {
  if (this.fileCounter >= xxx) {
    metadata.abort = true;
  } else {
     this.fileCounter++;
  }
  return metadata;
};

**Comment by olesea** [comment]
Publish it on npm, please

**Comment by Rehan1579** [comment]
facing same issue. Please update !

**Comment by meyer20** [comment]
+1

**Comment by emouawad** [comment]
Having same issue

![image](https://user-images.githubusercontent.com/6771272/48404876-ec651600-e739-11e8-9ecf-a9c6c527fe87.png)
